src/main/java/com/aws/greengrass/ipc/AuthorizeClientDeviceActionOperationHandler.java

```diff
@@ -0,0 +1,132 @@
+import static software.amazon.awssdk.aws.greengrass.GreengrassCoreIPCService.AUTHORIZE_CLIENT_DEVICE_ACTION;
+
+public class AuthorizeClientDeviceActionOperationHandler
+        extends GeneratedAbstractAuthorizeClientDeviceActionOperationHandler {
+    private static final Logger logger = LogManager.getLogger(AuthorizeClientDeviceActionOperationHandler.class);
+    private static final String COMPONENT_NAME = "componentName";
+    private static final String UNAUTHORIZED_ERROR = "Not Authorized";
+    private static final String NO_AUTH_TOKEN_ERROR = "Auth token is required";
+    private static final String NO_OPERATION_ERROR = "Mqtt operation is required";
+    private static final String NO_RESOURCE_ERROR = "Mqtt resource is required";
+    private final String serviceName;
+    private final AuthorizationHandler authorizationHandler;
+    private final DeviceAuthClient deviceAuthClient;
+
+    /**
+     * Constructor.
+     *
+     * @param context              operation continuation handler
+     * @param deviceAuthClient     device auth client
+     * @param authorizationHandler authorization handler
+     */
+    public AuthorizeClientDeviceActionOperationHandler(
+            OperationContinuationHandlerContext context,
+            DeviceAuthClient deviceAuthClient,
+            AuthorizationHandler authorizationHandler
+
+    ) {
+
+        super(context);
+        serviceName = context.getAuthenticationData().getIdentityLabel();
+        this.authorizationHandler = authorizationHandler;
+        this.deviceAuthClient = deviceAuthClient;
+    }
+
+    @SuppressWarnings({"PMD.AvoidCatchingGenericException", "PMD.PreserveStackTrace"})
+    @Override
+    public AuthorizeClientDeviceActionResponse handleRequest(AuthorizeClientDeviceActionRequest request) {
+        return translateExceptions(() -> {
+            try {
+                doAuthorizationForClientDevAction();
+            } catch (AuthorizationException e) {
+                logger.atWarn().kv("error", e.getMessage()).kv(COMPONENT_NAME, serviceName).log(UNAUTHORIZED_ERROR);
+                throw new UnauthorizedError(e.getMessage());
+            }
+            AuthorizationRequest authorizationRequest = getAuthzRequest(request);
+            try {
+                boolean isAuthorized = deviceAuthClient.canDevicePerform(authorizationRequest);
+                AuthorizeClientDeviceActionResponse response = new AuthorizeClientDeviceActionResponse();
+                return response.withIsAuthorized(isAuthorized);
+            } catch (InvalidSessionException e) {
+                logger.atError().log("Unable to find a valid session with the given auth token");
+                throw new InvalidClientDeviceAuthTokenError(
+                        "Unable to find a valid session with the given auth token. Check Greengrass log for details.");
+            } catch (Exception e) {
+                logger.atError().cause(e).log("Unable to authorize the client device action");
```

Comment:
This isn't really about not authorizing the client device action. It's a legitimate failure that shouldn't occur. Log something like "Unhandled exception while authorizing client device action" to make it more clear